sdk/eventhub/event-hubs/test/client.spec.ts

```diff
@@ -10,304 +10,374 @@ import chaiString from "chai-string";
-      void
-    > {
-      try {
-        client = new EventHubClient(service.connectionString, "bad" + Math.random());
-        await client.getPartitionProperties("0");
-      } catch (err) {
-        debug(err);
-        should.equal(err.name, "MessagingEntityNotFoundError");
-      }
-    });
+  it("should throw MessagingEntityNotFoundError while getting hub runtime info", async function(): Promise<void> {
+    try {
+      await client.getProperties();
+    } catch (err) {
+      debug(err);
+      should.equal(err.name, "MessagingEntityNotFoundError");
+    }
+  });
 
-    it("should throw MessagingEntityNotFoundError while creating a sender", async function(): Promise<void> {
-      try {
-        client = new EventHubClient(service.connectionString, "bad" + Math.random());
-        const sender = client.createProducer({ partitionId: "0" });
-        await sender.send([{ body: "Hello World" }]);
-      } catch (err) {
-        debug(err);
-        should.equal(err.name, "MessagingEntityNotFoundError");
-      }
-    });
+  it("should throw MessagingEntityNotFoundError while getting partition runtime info", async function(): Promise<void> {
+    try {
+      await client.getPartitionProperties("0");
+    } catch (err) {
+      debug(err);
+      should.equal(err.name, "MessagingEntityNotFoundError");
+    }
+  });
 
-    it("should throw MessagingEntityNotFoundError while creating a receiver", async function(): Promise<void> {
-      try {
-        client = new EventHubClient(service.connectionString, "bad" + Math.random());
-        const receiver = client.createConsumer(EventHubClient.defaultConsumerGroup, "0", EventPosition.earliest());
-        await receiver.receiveBatch(10, 5);
-      } catch (err) {
-        debug(err);
-        should.equal(err.name, "MessagingEntityNotFoundError");
-      }
-    });
+  it("should throw MessagingEntityNotFoundError while creating a sender", async function(): Promise<void> {
+    try {
+      const sender = client.createProducer({ partitionId: "0" });
+      await sender.send([{ body: "Hello World" }]);
+    } catch (err) {
+      debug(err);
+      should.equal(err.name, "MessagingEntityNotFoundError");
+    }
   });
 
-  describe("createConsumer", function(): void {
-    it("should throw an error if EventPosition is missing", function() {
-      try {
-        client = new EventHubClient(service.connectionString, service.path);
-        client.createConsumer(EventHubClient.defaultConsumerGroup, "0", undefined as any);
-        throw new Error("Test failure");
-      } catch (err) {
-        err.name.should.equal("TypeError");
-        err.message.should.equal(`Missing parameter "eventPosition"`);
-      }
-    });
+  it("should throw MessagingEntityNotFoundError while creating a receiver", async function(): Promise<void> {
+    try {
+      const receiver = client.createConsumer(EventHubClient.defaultConsumerGroup, "0", EventPosition.earliest());
+      await receiver.receiveBatch(10, 5);
```

Comment:
Do we need to call `receiveBatch`, or should the error occur when creating the consumer?

If we do need to call `receiveBatch`, should we be making sure to close the receiver once the test is completed?

Reply:
Only on calling an actual receive operation, the AMQP link will be set up which would then result in the error that we are testing. The actual "createConsumer" doesnt talk to the service

Reply:
`client.close()` is called at the end of each test, that will take care of closing all underlying receiver links